rbac/management/role/v2_model.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright 2025 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Model for role V2 management."""
+
+from uuid import uuid4
+
+from django.db import models
+from django.utils import timezone
+from management.models import Permission, Role
+from management.rbac_fields import AutoDateTimeField
+
+from api.models import TenantAwareModel
+
+
+class RoleV2(TenantAwareModel):
+    """A role."""
+
+    class Types(models.TextChoices):
+        CUSTOM = "custom"
+        SEEDED = "seeded"
+        PLATFORM = "platform"
+
+    uuid = models.UUIDField(default=uuid4, editable=False, unique=True, null=False)
+    name = models.CharField(max_length=150, null=False, blank=False)
+    display_name = models.CharField(max_length=150, default="")
+    description = models.TextField(null=True)
+    type = models.CharField(choices=Types.choices, max_length=20, null=False, blank=False, db_index=True)
```

Comment:
It is possible to create role without type, we can add validation or set default type (I guess custom? ) - I believe default is safest way.

Reply:
`default=None` (when combined with `null=False`) is also a reasonable option.

Reply:
yes default=custom seems to be a good idea